File: mobile/lib/data/model/ticket_request.dart
class TicketRequest {
  int seatId;
  String? seatClass;
  int flightId;
  int? luggageId;
  String name;
  String phone;
  String email;

  TicketRequest({
    required this.seatId,
    required this.seatClass,
    required this.flightId,
    required this.luggageId,
    required this.name,
    required this.phone,
    required this.email,
  });

  factory TicketRequest.fromJson(Map<String, dynamic> json) {
    return TicketRequest(
      seatId: json['seat_id'],
      seatClass: json['seatClass'],
      flightId: json['flight_id'],
      luggageId: json['luggage'],
      name: json['name'],
      phone: json['phone'],
      email: json['email'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'seatId': seatId,
      'seatCLass': seatClass,
      'flightId': flightId,
      'luggageId': luggageId,
      'name': name,
      'phone': phone,
      'email': email,
    };
  }
}

File: backend/flight_booking_project/src/main/java/com/javaweb/flight_booking_project/dto/request/TicketRequest.java
package com.javaweb.flight_booking_project.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.util.List;

@Data
public class TicketRequest {
    @NotNull(message = "Seat class not null")
    private String seatCLass;

    @NotNull(message = "Seat not null")
    private Long seatId;

    @NotNull(message = "Flight not null")
    private Long flightId;

    private Long luggageId;

    @NotNull(message = "Name not null")
    private String name;

    @NotNull(message = "Phone not null")
    private String phone;

    @NotNull(message = "Email not null")
    private String email;

}
